server/channels/api4/user.go

```diff
@@ -3338,6 +3354,10 @@ func updateReadStateAllThreadsByUser(c *Context, w http.ResponseWriter, r *http.
 		c.SetPermissionError(model.PermissionEditOtherUsers)
 		return
 	}
+	if !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), c.Params.TeamId, model.PermissionViewTeam) {
```

Comment:
Is this sufficient to make sure that only the threads that a user has subscribed to, are returned?

Reply:
@esarafianou same as above, since we are deleting thread memberships, the users cannot get back any that they are not subscribed.